### PR TITLE
Non restful items

### PR DESCRIPTION
**Pull request by pshim17**
Type of change

  - [X] New feature
  - [X] Bug Fix

Check the correct boxes

  - [X] This code broke nothing
  - [ ] This code broke some stuff
  - [ ] This code broke everything

Implements/Fixes:

  **Completed Find All Items by Name Fragment:**
  - Created find_all_items_spec: test passing 
  - Tests passing in Postman: 
     PASSED: happy path, fetch all items matching a pattern
     PASSED: sad path, no fragment matched 

  **Completed Find All Items by Price Fragment:**
  - Created find_all_items_spec: test passing 
  - Tests passing in Postman: 
      PASSED: happy path, fetch all items by min_price
      PASSED: sad path, cannot send name and min_price
      PASSED: sad path, min_price less than 0
      PASSED: sad path, max_price less than 0
      PASSED: happy path, min_price is so big that nothing matches
      PASSED: happy path, fetch all items by min and max price
      PASSED: sad path, cannot send name and max_price

  **Updated Find One Merchant By Name:**
   - Updated merchant search_controller to handle sad paths.
   - Tests weren't passing in Postman, they are passing now:
       - PASSED: happy path, fetch one merchant by fragment
       - PASSED: sad path, no fragment matched
   - Updated Find_a_merchant spec to test for sad path
       - spec tests PASSING
 
  **Extensions, edge cases:**
  - Find One Merchant by Fragment:
     - PASSED: edge case, no param given
     - PASSED: edge case, name fragment is empty
     
  - Find All Items by Fragment:
     - PASSED: edge case, no param given
     - PASSED: edge case, name fragment is empty

Testing Changes:

  - [x] I have fully tested my code 
  - [x] All tests are passing
  - [ ] Some tests are failing
  - [ ] All tests are failing

  - [X] No previous tests have been changed
  - [ ] Some previous tests have been changed
  - [ ] All of the previous tests have been changed (Please describe what in the world happened that all of the previous tests needed changing.)

Checklist:

  - [X] I have reviewed my code
  - [X] The code will run locally
  - [X] My code has no unused/commented out code
  - [ ] I have commented my code, particularly in hard-to-understand areas

Link: 

https://media3.giphy.com/media/v1.Y2lkPTc5MGI3NjExaG9jYndjZ2dsa3dxdTAwZTI2bmhqOTZrZmoyd3hiZDN0c2Z0OHI2ZyZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/7frSUXgbGqQPKNnJRS/giphy.webp

### CODE REVIEW

File: app/controllers/api/v1/items/search_controller.rb [hunk 1]
```diff
@@ -1,5 +1,8 @@
 class Api::V1::Items::SearchController < ApplicationController
   def show
-    item = Item.find_by_name(params[:name])
+    if params[:name]
+      items = Item.find_by_name(params[:name])
+    end
+    render json: ItemSerializer.new(items)
   end
-end
```
Comment (kevin-newland) [inline app/controllers/api/v1/items/search_controller.rb:5]: Great job using the search controller and item model to handle the database logic! I forgot to do this for get all merchants so I will be creating a new branch and moving the logic to keep it restful.

**Review by kevin-newland — APPROVED** [review]
Awesome work Patrick, your crushing it!

File: app/controllers/api/v1/invoices_controller.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+          invoice_status.push(invoice)
+        end
+      end
+      render json: InvoiceSerializer.new(invoice_status)
+    else
+      render json: InvoiceSerializer.new(invoices)
+    end
+  end
+
+  def merchant_not_found_error_response(error)
```
Comment (Jonathan-Atkins) [inline app/controllers/api/v1/invoices_controller.rb:21]: good job with rescue_from

**Review by Jonathan-Atkins — APPROVED** [review]
